tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,373 @@ def cvg_api(snappi_api_serv_ip,
+    """
+    prefix = 24
+    dutIps = create_ip_list("20.0.1.1", len(snappi_ports), mask=prefix)
+    tgenIps = create_ip_list("20.0.1.2", len(snappi_ports), mask=prefix)
+    ports = [port for port in snappi_ports if port['peer_device'] == duthost.hostname]
+
+    for port in ports:
+        port_id = port['port_id']
+        dutIp = dutIps[port_id]
+        tgenIp = tgenIps[port_id]
+        mac = __gen_mac(port_id)
+        logger.info('Configuring Dut: {} with port {} with IP {}/{}'.format(
+                                                                            duthost.hostname,
+                                                                            port['peer_port'],
+                                                                            dutIp,
+                                                                            prefix))
+        if port['asic_value'] == 'None':
+            duthost.command('sudo config interface ip add {} {}/{} \n' .format(
+                                                                                port['peer_port'],
+                                                                                dutIp,
+                                                                                prefix))
+        else:
+            duthost.command('sudo config interface -n {} ip add {} {}/{} \n' .format(
+                                                                                    port['asic_value'],
+                                                                                    port['peer_port'],
+                                                                                    dutIp,
+                                                                                    prefix))
+        device = config.devices.device(name='Device Port {}'.format(port_id))[-1]
+        ethernet = device.ethernets.add()
+        ethernet.name = 'Ethernet Port {}'.format(port_id)
+        ethernet.port_name = config.ports[port_id].name
+        ethernet.mac = mac
+        ip_stack = ethernet.ipv4_addresses.add()
+        ip_stack.name = 'Ipv4 Port {}'.format(port_id)
+        ip_stack.address = tgenIp
+        ip_stack.prefix = prefix
+        ip_stack.gateway = dutIp
+        port_config = SnappiPortConfig(
+                                        id=port_id,
+                                        ip=tgenIp,
+                                        mac=mac,
+                                        gw=dutIp,
+                                        gw_mac=str(duthost.facts['router_mac']),
+                                        prefix_len=prefix,
+                                        port_type=SnappiPortType.IPInterface,
+                                        peer_port=port['peer_port']
+                                      )
+        port_config_list.append(port_config)
+
+    return True
+
+
+def get_multidut_tgen_peer_port_set(line_card_choice, ports, config_set, number_of_tgen_peer_ports=2):
```

Comment:
tgen -> snappi

Reply:
this will addressed in a separate PR , an issue has been raised to track this - > https://github.com/sonic-net/sonic-mgmt/issues/8210